src/library/scala/collection/Factory.scala

```diff
@@ -298,6 +307,13 @@ trait StrictOptimizedSeqFactory[+CC[_]] extends SeqFactory[CC] {
     b.result()
   }
 
+  override def concat[A](xss: Iterable[A]*): CC[A] = {
+    val b = newBuilder[A]
+    b.sizeHint(xss.map(_.size).sum)
```

Comment:
`size` may not me constant time and require a collection traversal

Reply:
Ah I see this trait is inherited by `List` as well. I added the `.isInstanceOf[IndexedSeq[_]]` check that was present in 2.12. If there's a better way of doing it in 2.13 let me know.

Reply:
I think `knownSize` might be a better proxy

Reply:
Great! I updated to use `knownSize`.

Reply:
(updated)